test/SlowTests/Issues/RavenDB-19481.cs

```diff
@@ -0,0 +1,220 @@
+                Assert.Equal(1, stats.CountOfRevisionDocuments);
+
+                var restoredDatabaseName = $"restored_database-{Guid.NewGuid()}";
+                var backupPath = NewDataPath(suffix: "BackupFolder");
+                var config = Backup.CreateBackupConfiguration(backupPath);
+                RestoreBackupConfiguration restoreConfig;
+                if (options.DatabaseMode == RavenDatabaseMode.Sharded)
+                {
+                    var waitHandles = await Sharding.Backup.WaitForBackupToComplete(store);
+                    await Sharding.Backup.UpdateConfigurationAndRunBackupAsync(Server, store, config);
+                    Assert.True(WaitHandle.WaitAll(waitHandles, TimeSpan.FromMinutes(1)));
+                    var dirs = Directory.GetDirectories(backupPath);
+                    Assert.Equal(3, dirs.Length);
+                    var sharding = await Sharding.GetShardingConfigurationAsync(store);
+                    var settings = Sharding.Backup.GenerateShardRestoreSettings(dirs, sharding);
+                    restoreConfig = new RestoreBackupConfiguration { DatabaseName = restoredDatabaseName, ShardRestoreSettings = settings };
+                }
+                else
+                {
+                    var waitHandles = await Backup.WaitForBackupToComplete(store);
+                    await Backup.UpdateConfigAndRunBackupAsync(Server, config, store);
+                    restoreConfig = new RestoreBackupConfiguration { BackupLocation = Directory.GetDirectories(backupPath).First(), DatabaseName = restoredDatabaseName };
+                    Assert.True(WaitHandle.WaitAll(waitHandles, TimeSpan.FromMinutes(1)));
+                }
+
+                var clusterTransactions = new Dictionary<string, long>();
+                Server.ServerStore.ForTestingPurposesOnly().BeforeExecuteClusterTransactionBatch = (dbName, batch) =>
+                {
+                    if (dbName == restoredDatabaseName)
+                    {
+                        foreach (var clusterTx in batch)
+                        {
+                            var raftRequestId = clusterTx.Options.TaskId;
+                            if (clusterTransactions.ContainsKey(clusterTx.Options.TaskId) == false)
+                                clusterTransactions.Add(raftRequestId, 1);
+                            else
+                                clusterTransactions[raftRequestId]++;
+                        }
+                    }
+                };
+
+                using (Backup.RestoreDatabase(store, restoreConfig))
+                {
+                }
+
+                foreach (var kvp in clusterTransactions)
+                {
+                    var timesWasApplied = kvp.Value;
+                    Assert.True(timesWasApplied <= 1, $"cluster transaction \"{kvp.Key}\" was reapplied more then once ({timesWasApplied} times)");
```

Comment:
why `<=` don't we expect it to be exactly `1` ?

Reply:
changed it to `==`